src/sortStudents.ts

```diff
@@ -1,16 +1,54 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverage(grades: number[]): number {
+  return grades.reduce((acc, grade) => acc + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students]
+    .sort((studentA: Student, studentB: Student) => {
+      switch (sortBy) {
+        case SortType.Married:
+          return Number(studentB[sortBy]) - Number(studentA[sortBy]);
+
+        case SortType.Name:
+        case SortType.Surname:
+          return order === 'asc'
+            ? studentA[sortBy].localeCompare(studentB[sortBy])
+            : studentB[sortBy].localeCompare(studentA[sortBy]);
+
+        case SortType.Age:
+          return order === 'asc'
+            ? studentA[sortBy] - studentB[sortBy]
+            : studentB[sortBy] - studentA[sortBy];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+        case SortType.AverageGrade:
+          return order === 'asc'
+            ? getAverage(studentA[sortBy]) - getAverage(studentB[sortBy])
+            : getAverage(studentB[sortBy]) - getAverage(studentA[sortBy]);
+        default:
+          return 0;
+      }
```

Comment:
There is a better to throw an error, than 0
```suggestion
        default:
          throw new Error('Unknown sort type');
      }
```

Reply:
Fixed